mypy/nodes.py

```diff
@@ -2971,14 +2971,15 @@ def f(x: B[T]) -> T: ...  # without T, Any would be used here
         within functions that can't be looked up from the symbol table)
     """
     __slots__ = ('target', '_fullname', 'alias_tvars', 'no_args', 'normalized',
-                 'line', 'column', '_is_recursive', 'eager')
+                 'line', 'column', '_is_recursive', 'eager', 'rvalue')
 
     def __init__(self, target: 'mypy.types.Type', fullname: str, line: int, column: int,
                  *,
                  alias_tvars: Optional[List[str]] = None,
                  no_args: bool = False,
                  normalized: bool = False,
-                 eager: bool = False) -> None:
+                 eager: bool = False,
+                 rvalue: Optional[Expression] = None) -> None:
```

Comment:
You can try to make this just `Expression`. Similar constructs use `TempNode(type, no_rhs=True)`, when there's no `rvalue`

Reply:
Went in a different direction with this.